Null check IOSurfaceRef when creating WebCore::IOSurface from IOSurfaceRef

tuankiet65 opened the pull request:
#### a8fb29320ab86fe43d9ead073aeb8e8602f590a2
<pre>
Null check IOSurfaceRef when creating WebCore::IOSurface from IOSurfaceRef
<a href="https://bugs.webkit.org/show_bug.cgi?id=243241">https://bugs.webkit.org/show_bug.cgi?id=243241</a>

Reviewed by NOBODY (OOPS!).

When creating a WebCore::IOSurface from an IOSurfaceRef, WebCore::IOSurface expects
the IOSurfaceRef to be non-null, so it can pull out the dimension and allocation
size of the IOSurfaceRef.

* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::createFromSurface):
</pre>

youssefsoli commented on Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
```diff
@@ -213,6 +216,8 @@
     : m_colorSpace(colorSpace)
     , m_surface(surface)
 {
+    ASSERT(surface);
```
Is it possible for this assertion to be hit because surface null checked before it can reach this code? If so, an assertion might make more sense in the `::createFromSurface` method so it has the chance to be hit while in a debug build.

tuankiet65 replied:
I was going to say "defense mechanism hey!" but this is a private constructor already. I can remove this then. About asserting vs. returning `nullptr`, I like the latter more because `createFromImage` below does that.

youssefsoli replied:
Yup, it looks like `DisplayBufferDisplayDelegate` expects `createFromSurface` to potentially return a `nullptr` so it makes sense.

youssefsoli replied:
It might make sense to assert at the end of `DisplayBufferDisplayDelegate::setDisplayBuffer` in order to catch some simple errors, but I am not familiar with this code enough.

tuankiet65 committed:
Null check IOSurfaceRef when creating WebCore::IOSurface from IOSurfaceRef
https://bugs.webkit.org/show_bug.cgi?id=243241

Reviewed by NOBODY (OOPS!).

When creating a WebCore::IOSurface from an IOSurfaceRef, WebCore::IOSurface expects
the IOSurfaceRef to be non-null, so it can pull out the dimension and allocation
size of the IOSurfaceRef.

* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::createFromSurface):
